common/searchattribute/event_gen.go

```diff
@@ -0,0 +1,61 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package searchattribute
+
+import (
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	historypb "go.temporal.io/api/history/v1"
+)
+
+// TODO: Generate this file.
+
+func SetToEvent(event *historypb.HistoryEvent, sas *commonpb.SearchAttributes) {
+	switch event.GetEventType() {
+	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED:
+		event.GetWorkflowExecutionStartedEventAttributes().SearchAttributes = sas
```

Comment:
I'm always worried about a mismatch between the type enum and the oneof field.. if I understand correctly, if there is a mismatch, this will crash by assigning to a nil pointer?

Should it use `switch event.GetAttributes().(type)` instead?

Reply:
With codegen, it won't compile. Codegen gets event type from method name.

Reply:
I mean if it somehow got an event where EventType didn't match Attributes. Shouldn't happen but just to be defensive? I guess it's just a panic and some interceptor will catch it.

Reply:
so it is not possible for the same reason as with getter. But again, codegen now makes up event name from getter name. If they don't match, it will create non-existing event type and code won't compile.

Reply:
After second thought I came to idea that checking type of these internal properties would be a better idea. It is more defensive (as you said) and also less code.